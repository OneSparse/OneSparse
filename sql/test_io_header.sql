-- # Input/Output Operations
--
-- OneSparse comes with several functions to read and write different
-- text and binary formats for graph data.
--
-- ## SuiteSparse Binaries
--
-- SuiteSparse provides a very efficient compressed binary
-- representation for matrix and vector objects.

\set grbfile 'testmatrix.grb'

select serialize(random_matrix('int8', 10, 10, seed=>42)) as matrixbytes \gset
select deserialize(:'matrixbytes') = random_matrix('int8', 10, 10, seed=>42);

select serialize_file(random_matrix('int8', 10, 10, seed=>42), :'grbfile');
select deserialize_file(:'grbfile') = random_matrix('int8', 10, 10, seed=>42);

-- ## MatrixMarket Format
--
-- The Matrix Market format is a standard graph distribution format
-- used by the SuiteSparse Matrix Library.  OneSparse can read these
-- files and return matrix objects.

create materialized view if not exists karate
    as select mmread('/home/postgres/onesparse/demo/karate.mtx') as graph;

select print(graph) from karate;

-- ## Large Object Storage
--
-- Matrices bigger than the 1GB TOAST limit can be stored in the Large
-- Object table in Postgres, which supports storage up to 4TB.

select save(graph) as karate_loid from karate \gset

select print(load(:karate_loid));
