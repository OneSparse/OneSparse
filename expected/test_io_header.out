-- # IO Operations
--
-- select serialize(random_matrix(10, 10, 10, seed=>0.42, max=>99)) as matrixbytes \gset
\set grbfile 'testmatrix.grb'
select serialize(random_matrix(10, 10, 10, seed=>0.42, max=>99)) as matrixbytes \gset
ERROR:  function random_matrix(integer, integer, integer, seed => numeric, max => integer) does not exist
LINE 1: select serialize(random_matrix(10, 10, 10, seed=>0.42, max=>...
                         ^
HINT:  No function matches the given name and argument types. You might need to add explicit type casts.
select deserialize(:'matrixbytes') = random_matrix(10, 10, 10, seed=>0.42, max=>99);
ERROR:  syntax error at or near ":"
LINE 1: select deserialize(:'matrixbytes') = random_matrix(10, 10, 1...
                           ^
select serialize_file(random_matrix(10, 10, 10, seed=>0.42, max=>99), :'grbfile');
ERROR:  function random_matrix(integer, integer, integer, seed => numeric, max => integer) does not exist
LINE 1: select serialize_file(random_matrix(10, 10, 10, seed=>0.42, ...
                              ^
HINT:  No function matches the given name and argument types. You might need to add explicit type casts.
select deserialize_file(:'grbfile') = random_matrix(10, 10, 10, seed=>0.42, max=>99);
ERROR:  function random_matrix(integer, integer, integer, seed => numeric, max => integer) does not exist
LINE 1: select deserialize_file('testmatrix.grb') = random_matrix(10...
                                                    ^
HINT:  No function matches the given name and argument types. You might need to add explicit type casts.
